src/contract-testing/test-helpers/mock-server.ts

```diff
@@ -0,0 +1,105 @@
+import express, { Response } from 'express';
+import { handle } from '../../handlers/status-retriever-handler';
+import { ContextExamples } from '@aws-lambda-powertools/commons';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+
+let result: APIGatewayProxyResult;
+let server: any;
+export function setupServer(port: number) {
+  const app = express();
+
+  app.get('/ais/:userId', async (request, response) => {
+    const apiGatewayEvent = createDefaultApiRequest(request.params['userId']);
+    try {
+      result = await handle(apiGatewayEvent, ContextExamples.helloworldContext);
+      inspectStatus(result.statusCode, response);
+    } catch (error) {
+      determineError(error);
+    }
+  });
+
+  server = app.listen(port, () => {
+    console.log(`mock server listening on port ${port}`);
+  });
+}
+
+function inspectStatus(statusCode: number, response: Response) {
+  switch (statusCode) {
+    case 500: {
+      return response.status(500).json({ success: false, message: 'Internal Server Error.' });
+    }
+    default: {
+      response.send(JSON.parse(result.body));
+    }
+  }
+}
+
+function determineError(error: unknown) {
```

Comment:
I do not think this is needed, because all it does is log the error, regardless of the type of error..
What I would suggest is simply logging the error as an object + some text (like you do on line 44) and then gracefully return a 500 response with an error message that explains that call to handler was unsuccessful